File: can_make_arithmetic_progression_from_sequence.js
/**
 * @param {number[]} arr
 * @return {boolean}
 */
var canMakeArithmeticProgression = function(arr) {
    let minimum = Math.min(...arr);
    let maximum = Math.max(...arr);
    let n = arr.length;

    let diff = Math.floor((maximum - minimum) / (n - 1));
    let nums = new Set(arr);

    if (diff === 0) {
        return nums.size === 1;
    }

    for (let i = minimum; i <= maximum; i += diff) {
        if (!nums.has(i)) {
            return false;
        }
    }

    return true;    
};

File: can_make_arithmetic_progression_from_sequence.java
class Solution {
    public boolean canMakeArithmeticProgression(int[] arr) {
        int minimum = Integer.MAX_VALUE;
        int maximum = Integer.MIN_VALUE;
        int n = arr.length;

        for (int num : arr) {
            minimum = Math.min(minimum, num);
            maximum = Math.max(maximum, num);
        }

        int diff = (maximum - minimum) / (n - 1);
        Set<Integer> nums = new HashSet<>();

        for (int num : arr) {
            nums.add(num);
        }

        if (diff == 0) {
            return nums.size() == 1;
        }

        for (int i = minimum; i <= maximum; i += diff) {
            if (!nums.contains(i)) {
                return false;
            }
        }

        return true;
    }
}
